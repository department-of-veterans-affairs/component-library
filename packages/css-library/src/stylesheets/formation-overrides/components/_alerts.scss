// This is a copy of the entire ~uswds/src/stylesheets/components/alerts.scss file
// so that we can convert the rem values for USWDS v3.

/*
  Original: @import "~uswds/src/stylesheets/components/alerts";
  ***THIS IS USWDS v1***
*/

@import '../../override-function';

// Alert variables ---------- //
$usa-custom-alerts: () !default;
$usa-custom-alerts-bar: () !default;

$usa-alerts: (
  success: $vads-color-success-lighter,
  warning: $vads-color-gold-lightest,
  error: $vads-color-secondary-lightest,
  info: $vads-color-primary-alt-lightest
);

$usa-alerts-bar: (
  success: $vads-color-success-dark,
  warning: $vads-color-warning,
  error: $vads-color-secondary,
  info: $vads-color-info
);

$alerts: map-merge($usa-alerts, $usa-custom-alerts);
$alerts-bar: map-merge($usa-alerts-bar, $usa-custom-alerts-bar);

$h-padding: $spacing-medium;
$bar-size: $spacing-small;
$left-padding: $h-padding + $bar-size;
$icon-size: scale-rem(3rem);

.usa-alert {
  background-color: $color-gray-lightest;
  background-position: $h-padding $h-padding;
  background-repeat: no-repeat;
  background-size: scale(5.2rem); // TODO abrituary number based on svg size maybe?
  padding-bottom: $h-padding * 0.7;
  padding-left: $left-padding;
  padding-right: $h-padding;
  padding-top: $h-padding;
  position: relative;

  &::before {
    background-color: darken($color-gray-lightest, 40%);
    content: '';
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: $bar-size;
  }

  a {
    color: $color-primary-darker;

    &:focus,
    &:hover {
      color: $vads-color-primary-darker;
    }
  }

  ul {
    margin-bottom: 0;
    margin-top: $spacing-small;
    padding-left: $spacing-small;
  }
}

.usa-alert-slim {
  background-position: $h-padding $spacing-small;
  padding-bottom: $spacing-small;
  padding-top: $spacing-small;

  .usa-alert-text:only-child {
    margin-bottom: $spacing-x-small;
    padding-top: $spacing-x-small;
  }
}

.usa-alert-icon {
  display: table-cell;
  padding-right: $spacing-small;
}

.usa-alert-body {
  display: table-cell;
  vertical-align: top;
}

.usa-alert-heading {
  margin-top: 0;
  margin-bottom: 0;
}

.usa-alert-text {
  font-family: $font-family-sans;
  margin-bottom: 0;
  margin-top: 0;
}

.usa-alert-text:only-child {
  margin-bottom: $spacing-small;
  padding-top: $spacing-x-small;
}

@each $name, $bgcolor in $alerts {
  .usa-alert-#{$name} {
    background-color: $bgcolor;
    background-image: url('#{$image-path}/alerts/#{$name}.png');
    background-image: url('#{$image-path}/alerts/#{$name}.svg');

    &::before {
      background-color: map-get($alerts-bar, $name);
    }

    .usa-alert-body {
      padding-left: $icon-size + $h-padding;
    }
  }
}

.usa-alert-no_icon {
  background-image: none;
}

.usa-alert-paragraph {
  width: $text-max-width;
}

// Deprecated
.usa-alert {
  margin-top: 1.5em; // TODO deprecate
}
