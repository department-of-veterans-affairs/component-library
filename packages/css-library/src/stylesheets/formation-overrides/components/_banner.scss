// This is a copy of the entire ~uswds/src/stylesheets/components/banner.scss file
// so that we can convert the rem values for USWDS v3.

/*
  Original: @import "~uswds/src/stylesheets/components/banner";
  ***THIS IS USWDS v1***
*/

@import '../../override-function';

.usa-banner {
    background-color: $color-gray-lightest;
    padding-bottom: scale-rem(0.7rem);
  
    @include media($small-screen) {
      font-size: $h6-font-size;
      padding-bottom: 0;
    }
  }
  
  .usa-banner-content {
    @include padding(null $site-margins-mobile 3px $site-margins-mobile);
    background-color: transparent;
    font-size: $h5-font-size;
    padding-top: scale-rem(2rem);
    width: 100%;
  
    @include media($medium-screen) {
      padding-bottom: 23px;
      padding-top: 40px;
    }
  
    @include media($nav-width) {
      @include padding(null $site-margins);
    }
  
    p {
      &:first-child {
        margin-top: scale-rem(1rem);
  
        @include media($medium-screen) {
          margin-top: 0;
        }
      }
    }
  }
  
  .usa-banner-inner {
    @include outer-container($site-max-width);
    @include padding(null $site-margins-mobile);
  
    @include media($nav-width) {
      @include padding(null $site-margins);
    }
  }
  
  .usa-banner-header {
    @include padding(scale-rem(0.55rem) null);
    font-size: scale-rem(1.2rem);
    font-weight: $font-weight-normal;
  
    @include media($small-screen) {
      @include padding(0 null);
    }
  
    p {
      @include margin(0 null);
      display: inline;
      overflow: hidden;
      vertical-align: middle;
  
      @include media($small-screen) {
        @include margin(2px null);
        display: inline-block;
      }
    }
  
    img {
      float: left;
      margin-right: scale-rem(1rem);
      margin-top: 1px;
      width: scale-rem(2.4rem);
  
      @include media($small-screen) {
        margin-right: 7px;
        width: 20px;
      }
    }
  }
  
  .usa-banner-header-expanded {
    align-items: center;
    border-bottom: 1px solid $color-gray-light;
    display: flex;
    font-size: $h5-font-size;
    min-height: scale-rem(5.6rem);
    padding-right: scale-rem(3rem);
  
    @include media($small-screen) {
      border-bottom: none;
      display: block;
      font-size: scale-rem(1.2rem);
      font-weight: $font-weight-normal;
      min-height: 0;
      padding-right: 0;
    }
  
    > .usa-banner-inner {
      margin-left: 0;
  
      @include media($small-screen) {
        margin-left: auto;
      }
    }
  
    img {
      margin-right: scale-rem(2.4rem);
  
      @include media($small-screen) {
        margin-right: scale-rem(0.7rem);
      }
    }
  
    p {
      line-height: $heading-line-height;
      vertical-align: top;
  
      @include media($small-screen) {
        line-height: $base-line-height;
        vertical-align: middle;
      }
    }
  }
  
  .usa-banner-button {
    @include button-unstyled;
    @include padding(13px null null 48px);
    background-position-x: right;
    color: $color-primary;
    display: block;
    font-size: 12px;
    height: scale-rem(4.3rem);
    left: 0;
    position: absolute;
    text-decoration: underline;
    top: 0;
    width: 100%;
  
    @include media($small-screen) {
      @include padding(0 null null 0);
      /* stylelint-disable declaration-block-no-ignored-properties */
      display: inline;
      height: auto;
      margin-left: 3px;
      position: relative;
      vertical-align: middle;
      width: auto;
      /* stylelint-enable */
    }
  
    &:hover {
      color: $color-primary-darker;
      text-decoration: underline;
    }
  
    &[aria-expanded=false] {
      @include display-icon(angle-arrow-down-primary, after, scale-rem(1rem), 0, hover);
      background-image: none;
  
      &::after {
        background-position-y: 1px;
      }
    }
  
    &[aria-expanded=true] {
      @include display-icon(close, after, scale-rem(1.3rem), 0, no-hover);
      background-image: none;
      height: scale-rem(5.6rem);
  
      @include media($small-screen) {
        @include display-icon(angle-arrow-up-primary, after, scale-rem(1rem), 0, hover);
        height: initial;
        padding: 0;
        position: relative;
  
        &::after {
          background-position-y: 1px;
        }
      }
  
      &::after {
        position: absolute;
        right: scale-rem(1.5rem);
        top: scale-rem(2.15rem);
  
        @include media($small-screen) {
          position: static;
        }
      }
  
      .usa-banner-button-text {
        display: none;
  
        @include media($small-screen) {
          display: inline;
        }
      }
    }
  }
  
  .usa-banner-icon {
    width: 38px;
  }
  