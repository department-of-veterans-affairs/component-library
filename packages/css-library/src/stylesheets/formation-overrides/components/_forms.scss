// This is a copy of the entire ~uswds/src/stylesheets/components/forms.scss file
// so that we can convert the rem values for USWDS v3.

/*
  Original: @import "~uswds/src/stylesheets/components/forms";
  ***THIS IS USWDS v1***
*/
@use "../../core-variables" as core-vars;
@use "../../../../dist/tokens/scss/variables.scss" as tok;
@use "../variables" as fv;
@use '../../override-function' as *;
@use '../../mixins' as fm;

$usa-form-width: scale-rem(32rem);

[type=submit] {
  @include fm.media(tok.$medium-screen) {
    margin-top: scale-rule(calc(3rem - .2em)); // Label margin minus input bottom margin
  }
}

fieldset {
  border: none;
  margin: 0;
  padding: 0;
}

.usa-form {
  @include fm.media(tok.$medium-screen) {
    max-width: $usa-form-width;
  }
}

.usa-input-tiny {
  @include fm.media(tok.$medium-screen) {
    max-width: scale-rem(6rem);
  }
}

.usa-input-medium {
  @include fm.media(tok.$medium-screen) {
    max-width: scale-rem(12rem);
  }
}

.usa-form-note {
  float: right;
  font-family: tok.$font-family-sans;
  font-size: fv.$h5-font-size;
  margin: 0 0 scale(1.5rem);

  + * {
    clear: both;
  }
}

.usa-form-large {
  @include fm.media(tok.$medium-screen) {
    max-width: scale-rem(46rem);
  }
}

.usa-input-grid {
  @include fm.media(tok.$medium-screen) {
    padding-right: 5%;
  }
}

.usa-input-grid:last-of-type {
  @include fm.media(tok.$medium-screen) {
    padding-right: 0;
  }
}

.usa-input-grid-small {
  @include fm.media(tok.$medium-screen) {
    float: left;
    width: 35%;
  }

  input {
    margin-bottom: scale-rem(3rem);
  }

  select {
    margin-bottom: scale-rem(3rem);
  }
}

.usa-input-grid-medium {
  @include fm.media(tok.$medium-screen) {
    float: left;
    width: 65%;
  }

  input {
    margin-bottom: scale-rem(3rem);
  }

  select {
    margin-bottom: scale-rem(3rem);
  }
}

.usa-input-grid-large {
  @include fm.media(tok.$medium-screen) {
    float: left;
    width: 100%;
  }

  input {
    margin-bottom: scale-rem(3rem);
  }

  select {
    margin-bottom: scale-rem(3rem);
  }
}

.usa-additional_text {
  font-weight: normal;
}

// Reset password checklist

.usa-checklist {
  list-style: none;
  margin-left: 0;
  padding-left: 0;

  li {
    display: inline-block;
    list-style: none;
    margin-bottom: 0;
    margin-top: 0;
    padding-left: 3em;
    text-indent: -2em;

    &::before {
      content: ' ';
      display: inline-block;
      height: .8em;
      margin-right: .2em;
      width: 1.8em;
    }
  }
}

.usa-checklist-checked {
  &::before {
    background-image: url('#{core-vars.$image-path}/correct9.png');
    background-image: url('#{core-vars.$image-path}/correct9.svg');
    background-position: 100%;
    background-repeat: no-repeat;
    background-size: 100%;
  }
}
