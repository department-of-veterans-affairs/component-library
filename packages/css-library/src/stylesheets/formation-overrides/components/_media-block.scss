// This is a copy of the entire ~uswds/src/stylesheets/components/media-blocks.scss file
// so that we can convert the rem values for USWDS v3.

/*
  Original: @import "~uswds/src/stylesheets/components/media-block";
  ***THIS IS USWDS v1***
*/

@use '../../override-function' as *;

@mixin media-block-img($margin-right: scale-rem(1rem)) {
    float: left;
    margin-right: $margin-right;
  }
  
  .usa-media_block-img {
    @include media-block-img;
  }
  
  .usa-media_block-body {
    overflow: hidden;
  }
  