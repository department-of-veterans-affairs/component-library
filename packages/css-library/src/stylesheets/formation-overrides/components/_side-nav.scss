// This is a copy of the entire ~uswds/src/stylesheets/components/sidenav.scss file
// so that we can convert the rem values for USWDS v3.

/*
  Original: @import "~uswds/src/stylesheets/components/side-nav";
  ***THIS IS USWDS v1***
*/

@import '../../override-function';

.usa-sidenav-list {
    @include usa-sidenav-list;
    border-bottom: 1px solid;
    border-top: 1px solid;
  
    @include media($large-screen) {
      border-bottom: none;
      border-top: none;
    }
  
    .usa-grid & {
      @include margin(null (-$site-margins-mobile) null (-$site-margins-mobile));
  
      @include media($medium-screen) {
        @include margin(null 0);
      }
    }
  }
  
  .usa-sidenav-sub_list {
    @include usa-sidenav-sublist;
  }
  
  .usa-layout-docs-sidenav {
    padding-top: scale-rem(2.4rem);
  
    @include media($large-screen) {
      padding-top: 0;
    }
  }
  