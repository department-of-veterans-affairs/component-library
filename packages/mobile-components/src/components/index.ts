import { segmentedControl } from "./segmentedControl/segmentedControl";