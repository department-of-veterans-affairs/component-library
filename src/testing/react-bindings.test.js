// these imports are something you'd normally configure Jest to import for you
// automatically. Learn more in the setup docs: https://testing-library.com/docs/react-testing-library/setup#cleanup
import '@testing-library/jest-dom';
// NOTE: jest-dom adds handy assertions to Jest and is recommended, but not required

import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { VaTextInput } from '../../generated/bindings/components';

test('renders a binding for a Web Component', () => {
  const { container } = render(<VaTextInput label="First name"></VaTextInput>);

  expect(container).toMatchSnapshot();
});

/**
 * Skipping this because I'm having problems getting it to work
 */
test.skip('registers custom event handlers to Web Component', () => {
  const handlerMock = jest.fn();
  const { container } = render(
    <VaTextInput label="First name" onVaChange={handlerMock}></VaTextInput>,
  );
  fireEvent('vaChange', container.querySelector('va-text-input'));

  expect(handlerMock).toHaveBeenCalled();
});
