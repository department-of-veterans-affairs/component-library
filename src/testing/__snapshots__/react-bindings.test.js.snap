// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a binding for a Web Component 1`] = `
<div>
  <va-text-input
    label="First name"
  />
</div>
`;
